pkg/api/handlers/util.go

```diff
@@ -109,3 +112,42 @@ func nodePoolEqualsWithScaling(old, new models.NodePool) error {
 
 	return nil
 }
+
+func fetchOpenstackMetadata(request *http.Request, principal *models.Principal) (*models.OpenstackMetadata, error) {
+	tokenID := request.Header.Get("X-Auth-Token")
+
+	authOptions := &tokens.AuthOptions{
+		IdentityEndpoint: principal.AuthURL,
+		TokenID:          tokenID,
+		AllowReauth:      true,
```

Comment:
Reauth doesn't work for token based auth, once the token expires its game over.

Reply:
removed the `AllowReauth`